### PR TITLE
Training code

### PR DESCRIPTION
**Pull request by proteneer**
This PR sketches out the training cycle for a single edge. I'm running this overnight to see how the results hold up. This implements

- Training of charge and lennard jones parameters
- Illustrates usage of the vjp_fns to backprop system parameters into forcefield parameters
- Demonstrates it on a protein ligand complex
- Uses an extremely conservative and suboptimal clipped gradient optimizer.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,7 +44,7 @@ pytest -xsv tests/
 
 ## Theory
 
-The free energy difference is defined as the ratio of two partition functions:
+The free energy difference of transforming B into A is defined as the ratio of two partition functions:
```
Comment (maxentile) [inline README.md:47]:
```suggestion
The free energy difference of transforming B into A is defined as the log ratio of two partition functions:
```

**Comment by proteneer** [comment]
@maxentile I think this is mostly ready to go. Can you take a look when you get a chance?

**Review by maxentile — APPROVED** [review]
Nice -- looks great!

File: examples/rbfe_single.py [hunk 2]
```diff
@@ -30,6 +32,112 @@ def wrap_method(args, fn):
...
+
+    print("loss", loss, "pred", pred, "label", label)
+
+    dl_dpred = np.sign(pred - label)
+
+    # (ytz): these should be made configurable later on.
+    lrs = {
+        nonbonded.AM1CCCHandler: 0.05,
+        nonbonded.LennardJonesHandler: np.array([0.001,0])
+    }
```
Comment (maxentile) [inline examples/rbfe_single.py:112]: Minor variable-naming issue -- I would keep "learning rate" distinct from "gradient clipping threshold."

File: README.md (hunk 1 above)
Comment (proteneer) [reply README.md:47]: Great catch!

File: examples/rbfe_single.py (hunk 2 above)
Comment (proteneer) [reply examples/rbfe_single.py:112]: Agreed
